src/aotcompile.cpp

```diff
@@ -284,6 +295,7 @@ jl_code_instance_t *jl_ci_cache_lookup(const jl_cgparams_t &cgparams, jl_method_
     jl_value_t *ci = cgparams.lookup(mi, world, world);
     JL_GC_PROMISE_ROOTED(ci);
     jl_code_instance_t *codeinst = NULL;
+    JL_GC_PUSH1(&codeinst);
```

Comment:
This looks like a proper bugfix? Shouldn't the verifier have noticed this?

Reply:
Yes --- on this branch analyzegc complains about this, but on master it doesn't and I don't know why.

Reply:
IIRC the analyzer is heuristically driven, so the changes here must have changed how it visits the codebase.

Reply:
I don't see why this root is required.